cli/replicaset/configsource.go

```diff
@@ -106,6 +115,71 @@ func (c *CConfigSource) patchInstanceConfig(instanceName string, force bool,
 	return nil
 }
 
+// patchConfigWithRoles runs an config patching pipeline with adding roles.
+func (c *CConfigSource) patchConfigWithRoles(ctx RolesAddCtx,
+	getPathFunc func(clusterConfig libcluster.ClusterConfig,
+		ctx RolesAddCtx) (paths []path, err error),
+	patchFunc func(config *libcluster.Config, path []string,
+		roleNames []Role) (*libcluster.Config, error),
+) error {
+	configData, clusterConfig, err := collectCConfig(c.collector)
+	if err != nil {
+		return err
+	}
+	paths, err := getPathFunc(clusterConfig, ctx)
+	if err != nil {
+		return err
+	}
+
+	var target patchTarget
+	var patched *libcluster.Config
```

Comment:
I think we could add a linter for this case.
```suggestion
	var (
	     target patchTarget
	     patched *libcluster.Config
	)
```

Reply:
Unfortunately, I didn't find linter for this case.

Reply:
Ok, let's just apply the suggestion now.

Reply:
Applied.